src/Core/include/OpenLoco/Core/FileStream.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "Stream.hpp"
+#include <cstdint>
+#include <cstdio>
+#include <filesystem>
+#include <fstream>
+
+namespace OpenLoco
+{
+    class FileStream final : public Stream
+    {
+    private:
+        std::fstream _fstream;
+        StreamMode _mode{};
+        size_t _length{};
+        size_t _offset{};
+
+    public:
+        FileStream(const std::filesystem::path& path, StreamMode mode);
+
+        bool open(const std::filesystem::path& path, StreamMode mode);
```

Comment:
do we really need the constructor and the open command? Is this just for api reasons?

Reply:
Yeah I want to have this a bit more like fstream, I want to change the use later on, but I would like to avoid spilling more changes unless the API solidifies first.